graylog2-server/src/main/java/org/graylog2/lookup/caches/GuavaLookupCache.java

```diff
@@ -198,4 +215,90 @@ public static Builder builder() {
             public abstract Config build();
         }
     }
+
+    private static class InstrumentedCache<K, V> implements Cache<K, V> {
```

Comment:
I actually didn't know about the CacheStatsSet...

The problem here is that this would work for the Guava based cache, but not for others. The cache metrics are defined in the LookupCache class so every cache gets the same metrics. We rely on those to show the count and hit rate in the UI.